### DISCUSSION TITLE
Is this an error in the documentation on the example of the size of images in the Coordinate System?

### DISCUSSION THREAD

**Discussion by jj-jitterbit** [python-pillow/Pillow#7155] [Q&A] [answered]
In the docs, it says, for the [Coordinate System](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#coordinate-system):

> For example, a rectangle covering all of an 800x600 pixel image is written as (0, 0, 800, 600).

Is this not incorrect, as since the images are zero-based, that text (`(0, 0, 800, 600)`) should actually be for an image `801 x 601`? And so that line of text should really be:

> For example, a rectangle covering all of an 800x600 pixel image is written as (0, 0, 799, 599).

**Comment by radarhere** [comment]
Thanks. I've created PR #7156

**Reply by radarhere** [reply]
It was [pointed out](https://github.com/python-pillow/Pillow/pull/7156#issuecomment-1542959676) that while
```python
from PIL import Image, ImageDraw
im = Image.new("RGB", (801, 601))
d = ImageDraw.Draw(im)
d.rectangle((0, 0, 800, 600), "#f00")
im.show()
```
does produce a completely red image,
```python
from PIL import Image, ImageDraw
im = Image.new("RGB", (801, 601))
im.crop((0, 0, 800, 600)).show()
```
does not produce a (801, 600) image. While ImageDraw's `rectangle()` does include the endpoints, `crop()` does not.

Changing the behaviour of either of those methods would not be backwards compatible. To start talking about specific methods in the general "Coordinate system" part of the documentation would seem complicated. I'm trying again with #7169, removing the mention of behaviour of a specific rectangle relates to an image.

The [ImageDraw `rectangle()`](https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#PIL.ImageDraw.ImageDraw.rectangle) mentions that it includes the endpoints, to help those who are using that method.

**Reply by radarhere** [reply, marked as answer]
#7169 has now been merged. You can see the updated documentation at https://pillow.readthedocs.io/en/latest/handbook/concepts.html#coordinate-system, while we wait for the next Pillow version to be released.

Has that sufficiently explained things?
